subsys/compress/Kconfig

```
@@ -0,0 +1,100 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config NRF_COMPRESS
+	bool "nRF compression/decompression library"
+
+if NRF_COMPRESS
+
+config NRF_COMPRESS_COMPRESSION
+	bool
+	help
+	  Will enable support for compression functions in library.
+
+config NRF_COMPRESS_DECOMPRESSION
+	bool "Decompression support"
+	help
+	  Will enable support for decompression functions in library.
+
+menu "Type support"
+	depends on NRF_COMPRESS_COMPRESSION || NRF_COMPRESS_DECOMPRESSION
+
+config NRF_COMPRESS_TYPE_SELECTED
+	bool
+	help
+	  Hidden option to know if at least one implmentation has been selected
+
+menuconfig NRF_COMPRESS_LZMA
+	bool "LZMA"
+	depends on NRF_COMPRESS_DECOMPRESSION
+	select NRF_COMPRESS_TYPE_SELECTED
+	help
+	  Enables LZMA support for decompression.
+
+if NRF_COMPRESS_LZMA
+
+choice NRF_COMPRESS_LZMA_VERSION
+	prompt "Version"
+	default NRF_COMPRESS_LZMA_VERSION_LZMA2
+
+config NRF_COMPRESS_LZMA_VERSION_LZMA1
+	bool "lzma1"
+
+config NRF_COMPRESS_LZMA_VERSION_LZMA2
+	bool "lzma2"
+
+endchoice
+
+endif # NRF_COMPRESS_LZMA
+
+endmenu
+
+config NRF_COMPRESS_CHUNK_SIZE
+	int "Chunk size"
+	default 2048
+	help
+	  The chunk size is the default maximum amount of data that the decompress bytes available
+	  function will return that should be provided to the decompress function, it is not a
+	  strict requirement to follow this (unless the function returns a lower value, then data
+	  of that size must be provided)
```

This is very hard to read. Also, do you mean the `decompress_bytes` function?
```suggestion
	  The default maximum amount of data that the decompress_bytes() function will return.
	  The chunk size must be provided to the decompress function, unless the function returns a lower value.
```

---

The wording here is not correct, it is the decompress_bytes_needed() function and it just means that the user should not provide more than that value and ideally should provide that amount, but if they can't because it's the end of the full input (i.e. last part) or if they are streaming and not enough is available then they can provide less

---

Your explanation is way clearer than the description, so how about:
```suggestion
	  The chunk size is the default maximum amount of data that should be provided to the decompress_bytes_needed() function. The value is returned by the decompress bytes 
	  function that is available. Strive to provide the exact amount returned and not more. If not possible otherwise, for example because of streaming, a lower amount can be provided as well.
```
Note: I don't know what "the decompress bytes available function" means.